text/0617-rfc-stages.md

```diff
@@ -0,0 +1,321 @@
+
+## Motivation
+
+It can be difficult for both users and contributors of Ember to know the status of a new feature. Has it been approved? Is it a work in progress? What still needs to be done? Which version of Ember did it ship in?
+
+We can see that stages are needed when we consider the following:
+
+1. Community feedback that it was difficult to follow along with the progression of Ember Octane's features
+2. Challenges uncovered in the existing RFC Tracking process
+3. The long-standing, but stagnant goal of providing a public-facing status board
+4. Confusion about what it means for an RFC to be merged. There is a common myth that a merged RFC indicates that a feature will be in the next release, when instead it is a green flag for implementation work to begin.
+
+To understand the problem, it is helpful to think of the informal stages a merged RFC already can progress through: We have RFCs for which we have needed to indicate a change in direction ([MU](https://emberjs.github.io/rfcs/0143-module-unification.html)), RFCs that have needed to be clarified/replaced ([Named Blocks](https://emberjs.github.io/rfcs/0226-named-blocks.html)), RFCs waiting to be implemented, and RFCs whose features have been included in a stable release. Each of these scenarios was handled in a thoughtful manner, but they were difficult for users to follow along with.
+
+This proposal aims to build on the success of RFC 300, [RFC (Request for Comments) Process Update]([https://github.com/emberjs/rfcs/blob/master/text/0300-rfc-process-update.md](https://github.com/emberjs/rfcs/blob/master/text/0300-rfc-process-update.md)).
+
+## Detailed design
+
+### Current Process
+
+For details of the current RFC process, see the current [README of the RFCs repo](https://github.com/emberjs/rfcs/blob/2a7b9e8605a56277a4dd515bf8e192d9ac09c12f/README.md):
+
+### Stages
+
+A successful RFC can move through the following sequential stages. Whenever an RFC moves to the next stage, there is a PR to update it.
+
+| Stage| Description| Requires FCP to enter? |
+| -----| -----------|----- |
+| [0 - Proposed](#Proposed) | A proposal for a change to Ember or its processes that is offered for community and team evaluation. | no |
+| [1 - Exploring](#Exploring) | An RFC deemed worth pursuing but in need of refinement. | no |
+| [2 - Accepted](#Accepted) | A fully specified RFC. Waiting for or in the process of implementation. | yes |
+| [3 - Ready for Release](#Ready-for-Release) | The implementation of the RFC is complete, including learning materials. | yes |
+| [4 - Released](#Released) | The work is published. If it is codebase-related work, it is in a stable version of the relevant package(s). | no |
+| [5 - Recommended](#Recommended) | The feature/resource is recommended for general use. | yes |
```

Comment:
I don't see much benefit between Ready for Release, Released, and Recommended, tbh. Maybe I'm thinking too narrowly about only code things, but it seems that if a feature is shipped, no one is really going to check the RFC stage to see whether they should use it. A beta warning in JS console would go much further.

Another way to model this (again, limited to code things) is to implement in userland (via addon) and then incorporate into "core" to signal that it's recommended. I just can't see myself checking RFCs other than to check if the feature is released, and in what version.

Reply:
"Recommended" now has more detail. 

"Ready for Release" gives us a point at which all the teams and contributors can confirm the work is done and ready to be "Released" with stability guarantees. Once something is "Released", it's the same as it is now, it is stable. I don't think we'd ever expect the average user to check the RFC stage to decide whether to use a feature, but the stage will inform how the community talks about or promotes a feature. It's possible we could expose the stage in a user-friendly format in a dashboard.

For example, "Template-only components" could reach "Released" and be usable, with API documentation, but take some time to reach the point where the codemods were hardened, the guides and tutorial had all been updated to use them where appropriate. At that point, we have another PR & FCP for the community and teams to agree that the feature is "Recommended"